src/app/course-view/component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, OnInit } from '@angular/core';
+import { Router, ActivatedRoute, Params } from '@angular/router';
+import { Course } from '../course-list/course/component';
+import { YacsService } from '../services/yacs.service';
+
+
+@Component({
+  selector: 'course-view',
+  templateUrl: './component.html',
+  styleUrls: ['./component.scss']
+})
+export class CourseViewComponent implements OnInit {
+
+  courses : Course[] = [];
+
+  constructor (
+    private yacsService : YacsService,
+    private activatedRoute: ActivatedRoute) { }
+
+  getCourses (params: Params) {
+    let newParams : Object = {};
+    // add show_sections to params
+    Object.assign(newParams, params, { show_sections: true, show_periods: true }); // cannot directly modify params for some reason
```

Comment:
The goal is to add properties `show_sections` and `show_periods` to the param being passed into `yacsService`. But if I can't figure out a way to directly modify `params` argument without breaking some syntax rules.